### ISSUE TITLE
Error when selecting a file from list when the quick menu is focused

### ISSUE THREAD

**Issue by trimclain** [ThePrimeagen/harpoon#358]
Whenever I try to navigate to any of the files from the list when the quick menu is focused, I get the following error:
![image](https://github.com/ThePrimeagen/harpoon/assets/84108846/358d8090-4071-4ca5-80a9-365cfba949f8)
I suppose this wasn't intended, since the previous version of `harpoon` developed this habit of mine :smile:

**Comment by YujanSubedi** [comment]
it is caused by "save_on_toggle=true" settings:
go to "~/.local/share/nvim/site/pack/packer/start/harpoon/lua/harpoon", and change select_menu_item function on ui.lua for quick fix, u can just add "self.settings.save_on_toggle=false" at start of function and self.settings.save_on_toggle=true" before "self:close_menu()" if u want to use save_on_toggle all time but below change got the issue fixed for me.



function HarpoonUI:select_menu_item(options)
    local bool_sot = self.settings.save_on_toggle
    if bool_sot then
        self.settings.save_on_toggle=false
    end

    local idx = vim.fn.line(".")

    -- must first save any updates potentially made to the list before
    -- navigating
    local list = Buffer.get_contents(self.bufnr)
    self.active_list:resolve_displayed(list)

    self.active_list:select(idx, options)

    if bool_sot then
        self.settings.save_on_toggle=true
    end

    self:close_menu()
end

**Comment by YujanSubedi** [comment]
it is caused by "save_on_toggle=true" settings:
go to "~/.local/share/nvim/site/pack/packer/start/harpoon/lua/harpoon", and change select_menu_item function on ui.lua for quick fix, u can just add "self.settings.save_on_toggle=false" at start of function and self.settings.save_on_toggle=true" before "self:close_menu()" if u want to use save_on_toggle all time but below change got the issue fixed for me.
```lua
function HarpoonUI:select_menu_item(options)
local bool_sot = self.settings.save_on_toggle
if bool_sot then
self.settings.save_on_toggle=false
end

local idx = vim.fn.line(".")

-- must first save any updates potentially made to the list before
-- navigating
local list = Buffer.get_contents(self.bufnr)
self.active_list:resolve_displayed(list)

self.active_list:select(idx, options)

if bool_sot then
    self.settings.save_on_toggle=true
end

self:close_menu()

end
```

**Comment by YujanSubedi** [comment]
it is caused by "save_on_toggle=true" in settings, for quick fix

go to "~/.local/share/nvim/site/pack/packer/start/harpoon/lua/harpoon",
and change select_menu_item function on ui.lua,
u can just add "self.settings.save_on_toggle=false" at start of function
and self.settings.save_on_toggle=true" before "self:close_menu()"
if u want to use save_on_toggle all time
but below change got the issue fixed for me.


```lua
function HarpoonUI:select_menu_item(options)
    local bool_sot = self.settings.save_on_toggle
    if bool_sot then
        self.settings.save_on_toggle=false
    end

    local idx = vim.fn.line(".")

    -- must first save any updates potentially made to the list before
    -- navigating
    local list = Buffer.get_contents(self.bufnr)
    self.active_list:resolve_displayed(list)

    self.active_list:select(idx, options)

    if bool_sot then
    self.settings.save_on_toggle=true
    end

    self:close_menu()

    end
```

bool_sot sees your save_on_toggle option and if u have it on temporary disables it
them once u select the file it enables it before closing

**Comment by YujanSubedi** [comment]
There is issue with save_on_toggle settings.

for quick fix change to "select_menu_item" function on
    "~/.local/share/nvim/site/pack/packer/start/harpoon/lua/harpoon/ui.lua"


```lua
function HarpoonUI:select_menu_item(options)
    -- is_sot_enabled is find save_on_toggle option
    local is_sot_enabled = self.settings.save_on_toggle
    -- if save_on_toggle is enabaled the temporarily disable it
    if is_sot_enabled then
        self.settings.save_on_toggle=false
    end

    local idx = vim.fn.line(".")

    -- must first save any updates potentially made to the list before
    -- navigating
    local list = Buffer.get_contents(self.bufnr)
    self.active_list:resolve_displayed(list)

    self.active_list:select(idx, options)

    -- enable save_on_toggle if it was enabled on settings
    if is_sot_enabled then
        self.settings.save_on_toggle=false
    end

    self:close_menu()
end
```

**Comment by YujanSubedi** [comment]
There is issue with save_on_toggle settings.

for quick fix change to "select_menu_item" function on
    "~/.local/share/nvim/site/pack/packer/start/harpoon/lua/harpoon/ui.lua"


```lua
function HarpoonUI:select_menu_item(options)
    -- is_sot_enabled is find save_on_toggle option
    local is_sot_enabled = self.settings.save_on_toggle
    -- if save_on_toggle is enabaled then temporarily disable it
    if is_sot_enabled then
        self.settings.save_on_toggle=false
    end

    local idx = vim.fn.line(".")

    -- must first save any updates potentially made to the list before
    -- navigating
    local list = Buffer.get_contents(self.bufnr)
    self.active_list:resolve_displayed(list)

    self.active_list:select(idx, options)

    -- enable save_on_toggle if it was enabled on settings
    if is_sot_enabled then
        self.settings.save_on_toggle=false
    end

    self:close_menu()
end
```

**Comment by ThePrimeagen** [comment]
This was supported in the previous version, and I thought this was supported in the current version.

Tell me your exact steps to reproduce?

It also seems like I need to get a logger in place such that you could provide me the exact calls internally

**Comment by trimclain** [comment]
> Tell me your exact steps to reproduce?

Using the [example config](https://github.com/ThePrimeagen/harpoon#basic-setup)
1. Open neovim
2. If there is no file in the list, add a file with `<leader>a`
3. Open harpoon quickmenu with `<C-e>`
4. Navigate to the file without closing the quickmenu with `<C-h>`
5. Get the error

**Comment by ThePrimeagen** [comment]
ok, let me look at this today

**Comment by ThePrimeagen** [comment]
i cannot seem to repro this

so i added a logger, could you please get the latest harpoon2? then follow the steps if the issue persists and paste in the log?

https://github.com/ThePrimeagen/harpoon/tree/harpoon2#logger

**Comment by trimclain** [comment]
Sure, here's the screenshot:

![image](https://github.com/ThePrimeagen/harpoon/assets/84108846/b0722b74-d567-4340-bc76-6e10aa8ee161)

And here's the message itself:
```
{ '"ui#toggle_quick_menu" "__harpoon_files"', '"ui#toggle_quick_menu" "__harpoon_files"', '"ui#save" { "nvim/.config/nvim/init.lua", "nvim/.config/nvim/lua/plugins/editor.l
ua" }', '"ui#close_menu name: " "__harpoon_files" "win and bufnr" {\n  bufnr = 13,\n  win = 1010\n}', '"toggle by BufLeave"', '"ui#toggle_quick_menu" nil', '"ui#save" { "nv
im/.config/nvim/init.lua", "nvim/.config/nvim/lua/plugins/editor.lua" }', '"ui#toggle_quick_menu" "__harpoon_files"', '"config_default#select" {\n  context = {\n    col = 8
,\n    row = 753\n  },\n  value = "nvim/.config/nvim/lua/plugins/editor.lua"\n} "__harpoon_files" nil', '"toggle by BufLeave"', '"ui#toggle_quick_menu" nil', '"ui#save" { "
nvim/.config/nvim/init.lua", "nvim/.config/nvim/lua/plugins/editor.lua" }', '"ui#close_menu name: " "__harpoon_files" "win and bufnr" {\n  bufnr = 18,\n  win = 1015\n}', '"
toggle by BufLeave"', '"ui#toggle_quick_menu" nil' }
E5108: Error executing lua ...in/.local/share/nvim/lazy/harpoon/lua/harpoon/logger.lua:36: 'replacement string' item contains newlines
stack traceback:
        [C]: in function 'nvim_buf_set_lines'
        ...in/.local/share/nvim/lazy/harpoon/lua/harpoon/logger.lua:36: in function 'show'
        [string ":lua"]:1: in main chunk
```
Also, I don't know if it helps, but I'm on `NVIM v0.10.0-dev-1223+gde9348978`

**Comment by ThePrimeagen** [comment]
I must be getting an error here. It should be opening up a nice buffer filled with all the log lines.

Let me take a look at this in a bit

**Comment by trimclain** [comment]
I saw you fixed the logger. So here's the new message I get:

```
ui#toggle_quick_menu#opening __harpoon_files
ui#toggle_quick_menu#closing __harpoon_files
ui#save {}
ui#close_menu name: __harpoon_files win and bufnr { bufnr = 6, win = 1005 }
toggle by BufLeave
ui#toggle_quick_menu#closing
ui#save {}
config_default#create_list_item init.lua
HarpoonList:append { index = -1, item = { context = { col = 56, row = 44 }, value = "init.lua" } }
config_default#create_list_item lua/plugins/editor.lua
HarpoonList:append { index = -1, item = { context = { col = 8, row = 771 }, value = "lua/plugins/editor.lua" } }
config_default#select { context = { col = 56, row = 44 }, value = "init.lua" } __harpoon_files
config_default#select { context = { col = 8, row = 771 }, value = "lua/plugins/editor.lua" } __harpoon_files
ui#toggle_quick_menu#opening __harpoon_files
config_default#select { context = { col = 56, row = 44 }, value = "init.lua" } __harpoon_files
toggle by BufLeave
ui#toggle_quick_menu#closing
ui#save { "init.lua", "lua/plugins/editor.lua" }
ui#close_menu name: __harpoon_files win and bufnr { bufnr = 24, win = 1022 }
toggle by BufLeave
ui#toggle_quick_menu#closing
```

I also recorded a video:

https://github.com/ThePrimeagen/harpoon/assets/84108846/c56eca31-a580-44ba-86f2-a7771d852435

Here is the config I'm using in this video:
```
require("harpoon"):setup({
    settings = { save_on_toggle = true }
})
vim.keymap.set("n", "<leader>a", ':lua require("harpoon"):list():append()<cr>')
vim.keymap.set("n", "<C-e>", ':lua require("harpoon").ui:toggle_quick_menu(require("harpoon"):list())<cr>')

vim.keymap.set("n", "<C-j>", ':lua require("harpoon"):list():select(1)<cr>')
vim.keymap.set("n", "<C-k>", ':lua require("harpoon"):list():select(2)<cr>')
```
PS: I also just realized, that in the comment above I lied to you about the config I was using :sweat_smile:
I forgot I set `save_on_toggle = true`. This is probably why you couldn't reproduce the error. Sorry :smile:

**Comment by ThePrimeagen** [comment]
i think that this has been solved by willothy's change #392 

please reopen if this is still an issue
